tools/profile_endpoints.py

```diff
@@ -0,0 +1,315 @@
+
+
+# Endpoint Profiling Functions
+def profile_get_snapshot_builder_count(token):
+    """
+    Profiles the getSnapshotBuilderCount endpoint and returns the time taken.
+
+    Args:
+        token (str): The access token for authentication.
+
+    Returns:
+        str: A string indicating the average time taken for the request.
+    """
+    print("Profiling getSnapshotBuilderCount endpoint")
+
+    # Define the bodies for the POST request
+    bodys = [
+        {
+            "cohorts": [
+                {
+                    "name": "",
+                    "criteriaGroups": [
+                        {
+                            "name": "Group 1",
+                            "mustMeet": True,
+                            "meetAll": False,
+                            "criteria": [
+                                {"kind": "range", "id": 0, "low": 1955, "high": 2021},
+                                {"kind": "list", "id": 2, "values": [38003563]},
+                                {"kind": "list", "id": 1, "values": [8507]},
+                            ],
+                        }
+                    ],
+                }
+            ]
+        },
+        {
+            "cohorts": [
+                {
+                    "name": "",
+                    "criteriaGroups": [
+                        {
+                            "name": "Group 1",
+                            "mustMeet": True,
+                            "meetAll": False,
+                            "criteria": [
+                                {"kind": "domain", "id": 19, "conceptId": 4042140},
+                                {"kind": "domain", "id": 19, "conceptId": 441840},
+                                {"kind": "domain", "id": 27, "conceptId": 46234754},
+                                {"kind": "list", "id": 3, "values": [8515]},
+                            ],
+                        }
+                    ],
+                }
+            ]
+        },
+        {
+            "cohorts": [
+                {
+                    "name": "",
+                    "criteriaGroups": [
+                        {
+                            "name": "Group 1",
+                            "mustMeet": True,
+                            "meetAll": False,
+                            "criteria": [
+                                {"kind": "range", "id": 0, "low": 1992, "high": 2021},
+                                {"kind": "list", "id": 1, "values": [0]},
+                                {"kind": "list", "id": 2, "values": [38003563]},
+                                {"kind": "list", "id": 3, "values": [8527]},
+                            ],
+                        }
+                    ],
+                }
+            ]
+        },
+    ]
+
+    time_length = 0
+    for body in bodys:
+        time_taken = make_post_request("count", token, body)
+        time_length += time_taken
+
+    return f"Time taken: {time_length / len(bodys)} seconds"
```

Comment:
This is averaging the times of three different calls to the same API. While it would make sense that a change that improves one operation would improve all of them, I think it would be better to split out the different operations as separate times, in case performance changes differently for each. You can label them as A B C or something since the specifics don't matter too much as long as they're consistent across different benchmarks.

Reply:
Ok, sounds good!